src/main/java/Chatbot.java

```diff
@@ -0,0 +1,398 @@
+import java.io.FileNotFoundException;
+import java.time.LocalDateTime;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.format.DateTimeFormatter;
+
+public class Chatbot {
+
+    public static class Task {
+        private String description;
+        private boolean done;
+
+        public Task(String description) {
+            this.description = description;
+            this.done = false;
+        }
+
+        public void markAsDone() {
+            this.done = true;
+        }
+
+        public void markAsNotDone() {
+            this.done = false;
+        }
+
+        public String getDescription() {
+            return description;
+        }
+
+        public boolean isDone() {
+            return done;
+        }
+    }
+
+    public static class Todo extends Task {
+        public Todo(String description) {
+            super(description);
+        }
+    }
+
+    public static class Deadline extends Task {
+        private LocalDateTime deadline;
+
+        String datePattern = "MMM-dd-yyyy HH:mm";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(datePattern);
+
+        public Deadline(String description, LocalDateTime deadline) {
+            super(description);
+            this.deadline = deadline;
+        }
+
+        public String getDescription() {
+            return super.description + " (by: " + deadline.format(formatter) + ")";
+        }
+
+        public String getDeadline() {
+            return deadline.format(formatter);
+        }
+
+        public String getDescriptionWithoutTime() {
+            return super.description;
+        }
+    }
+
+    public static class Event extends Task {
+        private LocalDateTime startTime;
+        private LocalDateTime endTime;
+
+        String datePattern = "MMM-dd-yyyy HH:mm";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(datePattern);
+
+        public Event(String description, LocalDateTime startTime, LocalDateTime endTime) {
+            super(description);
+            this.startTime = startTime;
+            this.endTime = endTime;
+        }
+
+        public String getDescription() {
+            String datePattern = "MMM-dd-yyyy HH:mm";
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(datePattern);
+            return super.description + " (from: " + startTime.format(formatter) + " to: "
+                    + endTime.format(formatter) + ")";
+        }
+
+        public String getStartTime() {
+            return startTime.format(formatter);
+        }
+
+        public String getEndTime() {
+            return endTime.format(formatter);
+        }
+
+        public String getDescriptionWithoutTime() {
+            return super.description;
+        }
+    }
+
+    public static class Ui {
+        public static void startMessage() {
```

Comment:
Perhaps a more descriptive method name here? I like showStartMessage() since the method has side effects like printing to the screen.